### PR TITLE
Convert canvas-particles to TS

### PR DESCRIPTION
**Pull request by SerG-Y**
#### Description of change
Converted canvas-particles to TS

##### Pre-Merge Checklist

- [x] Lint process passed (`npm run lint`)

### CODE REVIEW

File: packages/canvas/canvas-particles/src/ParticleContainer.ts [hunk 1]
```diff
@@ -32,7 +33,7 @@ ParticleContainer.prototype.renderCanvas = function renderCanvas(renderer)
 
     for (let i = 0; i < this.children.length; ++i)
     {
-        const child = this.children[i];
+        const child: any = this.children[i];
```
Comment (bigtimebuddy) [inline packages/canvas/canvas-particles/src/ParticleContainer.ts:36]: What about using Sprite here instead of any?

Comment (SerG-Y) [reply packages/canvas/canvas-particles/src/ParticleContainer.ts:36]: Yes, it's a good idea

Comment (SerG-Y) [reply packages/canvas/canvas-particles/src/ParticleContainer.ts]: Fixed

File: packages/canvas/canvas-particles/src/ParticleContainer.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 import { ParticleContainer } from '@pixi/particles';
+import { Sprite } from '@pixi/sprite';
```
Comment (bigtimebuddy) [inline packages/canvas/canvas-particles/src/ParticleContainer.ts:2]:
```suggestion
import type { Sprite } from '@pixi/sprite';
```

**Pull request merged**